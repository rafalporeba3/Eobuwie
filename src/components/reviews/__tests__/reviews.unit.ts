import { createLocalVue, Wrapper, WrapperArray } from '@vue/test-utils';
import Vuex                                      from 'vuex';
import { createTestInstanceShallow }             from '@/__tests__/factory';
import CompositionApi                            from '@vue/composition-api';
import cStarIcon                                 from '@/components/icons/star.component.vue';
import cReviews                                  from '../reviews.component.vue';

const localVue = createLocalVue();
localVue.use(CompositionApi);
localVue.use(Vuex);

describe('Reviews component', () => {
  let wrapper!: Wrapper<any>;

  beforeEach(() => {
    const {
      wrapperInstance,
    } = createTestInstanceShallow(cReviews, {
      localVue,
      propsData: {
        rating: 3.5,
        reviews: 123,
      },
    });
    wrapper = wrapperInstance;
  });

  it('should render cReviews component with proper class', () => {
    const reviewsComponent: Wrapper<any> = wrapper.findComponent(cReviews);

    expect(reviewsComponent.exists())
      .toBeTruthy();

    expect(reviewsComponent.classes())
      .toContain('c-reviews');
  });

  it('should get and display correct reviews', () => {
    const reviewsComponent: Wrapper<any> = wrapper.findComponent(cReviews);
    const reviewsCountElement: Wrapper<any> = wrapper.find('.c-reviews__count');

    expect(reviewsComponent.props('reviews'))
      .toBe(123);
    expect(reviewsCountElement.text())
      .toBe('123');
  });

  it('should display correct number of full stars', () => {
    const starComponents: WrapperArray<any> = wrapper.findAllComponents(cStarIcon);

    expect(starComponents.length)
      .toEqual(5);
  });

  it('should match snapshot', () => {
    expect(wrapper.html())
      .toMatchSnapshot();
  });
});
