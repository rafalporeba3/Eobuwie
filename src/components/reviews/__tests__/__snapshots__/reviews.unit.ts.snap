// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews component should match snapshot 1`] = `
<div class="c-reviews">
  <c-star-icon-stub isfullstar="true"></c-star-icon-stub>
  <c-star-icon-stub isfullstar="true"></c-star-icon-stub>
  <c-star-icon-stub isfullstar="true"></c-star-icon-stub>
  <c-star-icon-stub ishalfstar="true"></c-star-icon-stub>
  <c-star-icon-stub></c-star-icon-stub> <span class="c-reviews__count">123</span>
</div>
`;
